docs/source-app/glossary/app_checkpointing.rst

```diff
@@ -0,0 +1,113 @@
+:orphan:
+
+#############
+Checkpointing
+#############
+
+Beta feature.
+
+----
+
+*********************
+What is a checkpoint?
+*********************
+When a lightning app is running it is possible to save the state of the app and start it again from the saved checkpoint.
+This is useful for long running apps that need to be stopped and restarted or while developing an app and needing to recreate the state of the app to to continue development at a specific point.
+
+
+************************
+How to save a checkpoint
+************************
+
+Saving a checkpoint is simple. In main flow if your lightning app, implement `should_save_checkpoint()` that returns True when you want to save a checkpoint.
+
+.. code-block:: python
+
+    class MyLightningApp(L.LightningFlow):
+        def run(self):
+            # do stuff
+            pass
+
+        def should_save_checkpoint(self):
+            return True
+
+When `should_save_checkpoint()` returns True, Lightning will save the state of the app to a checkpoint file. The checkpoint file will be saved in lightning shared storage at `/checkpoints` and will be named `lightningapp_checkpoint_<timestamp>.json`.
```

Comment:
```suggestion
When ``should_save_checkpoint()`` returns ``True``, Lightning will save the state of the app to a checkpoint file. The checkpoint file will be saved in Lightning shared storage at ``/checkpoints`` and will be named ``lightningapp_checkpoint_<timestamp>.json``.
```

Reply:
Also, /checkpoints (root) is not correct, right? It probably means ./checkpoints (cwd)?

Reply:
Yes, I would say ``lit://checkpoints`

"The checkpoint file will be saved in Lightning [Drive Storage](https://lightning.ai/docs/stable/glossary/storage/drive.html) at `lit://checkpoints` "